src/components/App.js

```diff
@@ -152,103 +178,108 @@ var App = React.createClass({
 
     return channels
   },
-  _saveChannels: function(networkName, username, channels) {
+  _saveChannels: function (networkName, username, channels) {
     const channelsKey = this._makeChannelsKey(username, networkName)
     localStorage.setItem(channelsKey, JSON.stringify(channels))
   },
-  _showConnectView: function() {
+  _showConnectView: function () {
     this.setState({ user: null })
-    AppActions.setLocation("Connect")
+    AppActions.setLocation('Connect')
   },
-  onUserUpdated: function(user) {
-    logger.debug("User updated", user)
+  onUserUpdated: function (user) {
+    logger.debug('User updated', user)
 
     if (!user) {
-      AppActions.setLocation("Connect")
+      AppActions.setLocation('Connect')
       return
     }
 
-    if (user === this.state.user)
-      return
+    if (user === this.state.user) return
 
     this.setState({ user: user })
 
     if (!this.state.panelOpen) this.openPanel()
     AppActions.setLocation(null)
   },
-  joinChannel: function(channelName, password) {
+  joinChannel: function (channelName, password) {
     if (channelName === AppStateStore.state.currentChannel) {
       this.closePanel()
       return
     }
-    logger.debug("Join channel #" + channelName)
+    logger.debug('Join channel #' + channelName)
     NetworkActions.joinChannel(channelName, password)
   },
-  onJoinChannelError: function(channel, err) {
-    if(!this.state.panelOpen) this.setState({ panelOpen: true })
-    this.setState({ joiningToChannel: channel, requirePassword: true} )
+  onJoinChannelError: function (channel, err) {
+    if (!this.state.panelOpen) this.setState({ panelOpen: true })
+    this.setState({ joiningToChannel: channel, requirePassword: true })
   },
-  onJoinedChannel: function(channel) {
-    logger.debug("Joined channel #" + channel)
+  onJoinedChannel: function (channel) {
+    logger.debug('Joined channel #' + channel)
     this.closePanel()
     document.title = `#${channel}`
-    logger.debug("Set title: " + document.title)
+    logger.debug('Set title: ' + document.title)
     AppActions.setCurrentChannel(channel)
-    let channels = this._getSavedChannels(this.state.networkName, this.state.user.name)
+    let channels = this._getSavedChannels(
+      this.state.networkName,
+      this.state.user.name
+    )
     if (!channels.find(e => e.name === channel)) {
       channels.push({ name: channel })
       this._saveChannels(this.state.networkName, this.state.user.name, channels)
     }
   },
-  onLeaveChannel: function(channel) {
+  onLeaveChannel: function (channel) {
     const { user, networkName } = this.state
     const channelsKey = this._makeChannelsKey(user.name, networkName)
-    const channels = this._getSavedChannels(networkName, user.name).filter((c) => c.name !== channel)
-    if (channels.length === 0)
-      localStorage.removeItem(channelsKey)
-    else
+    const channels = this._getSavedChannels(networkName, user.name).filter(
```

Comment:
Stylistic nitpicking: I feel it'd be better to split this per function call, eg:
```javascript
const channels = this._getSavedChannels(networkName, user.name)
  .filter(c => c.name !== channel)
```